lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -92,7 +92,9 @@ def group_name
 
     def run_kubectl(*args)
       raise FatalDeploymentError, "Namespace missing for namespaced command" if namespace.blank?
-      args = args.unshift("kubectl").push("--namespace=#{namespace}")
+      raise FatalDeploymentError, "Explicit context is required to run this command" if context.blank?
+      args = args.unshift("kubectl").push("--namespace=#{namespace}").push("--context=#{context}")
```

Comment:
In general I agree with you that mutation should be avoided. In this case, however, given the purpose of this array, I'm having trouble imagining a scenario where getting the command that actually ran instead of the command passed in would be a bad thing. If, for example, you're trying to log it, it would actually be more accurate.